projects/wave-core-new/src/lib/datasets/dataset.model.ts

```diff
@@ -75,18 +77,19 @@ export class InternalDatasetId {
 
     toDict(): DatasetIdDict {
         return {
-            internal: this.id,
+            type: 'internal',
+            dataset: this.id,
         };
     }
 }
 
 export class ExternalDatasetId {
     provider: UUID;
-    id: string;
+    dataset: string;
```

Comment:
```suggestion
    id: string;
```

Reply:
I think `dataset` makes more sense here, because `povider` is also actually an id. Plus, the id is stored in an `id` attribute so it would be like `id.id`

Reply:
now you have 
```{
  "dataset": {
    "dataset": …
  }
}
```